internal/versions/versions.go

```diff
@@ -175,42 +175,103 @@ func (h *handler) GetDefaultReleaseImage(cpuArchitecture string) (*models.Releas
+		// Empty implies default CPU architecture
+		cpuArchitecture = common.DefaultCPUArchitecture
+	}
+	// Filter OS images by specified CPU architecture
+	osImages := funk.Filter(h.osImages, func(osImage *models.OsImage) bool {
+		if swag.StringValue(osImage.CPUArchitecture) == "" {
+			return cpuArchitecture == common.DefaultCPUArchitecture
+		}
+		return swag.StringValue(osImage.CPUArchitecture) == cpuArchitecture
+	})
+	if funk.IsEmpty(osImages) {
+		return nil, errors.Errorf("The requested CPU architecture (%s) isn't specified in OS images list", cpuArchitecture)
+	}
+
+	// Search for specified x.y.z openshift version
+	osImage := funk.Find(osImages, func(osImage *models.OsImage) bool {
+		return swag.StringValue(osImage.OpenshiftVersion) == openshiftVersion
+	})
+
 	versionKey, err := h.getKey(openshiftVersion)
 	if err != nil {
 		return nil, err
 	}
 
-	for _, osImage := range h.osImages {
-		if cpuArchitecture == "" {
-			// Empty implies default CPU architecture
-			cpuArchitecture = common.DefaultCPUArchitecture
-		}
-		if *osImage.OpenshiftVersion == versionKey && *osImage.CPUArchitecture == cpuArchitecture {
-			return osImage, nil
+	if osImage == nil {
+		// Fallback to x.y version
+		osImage = funk.Find(osImages, func(osImage *models.OsImage) bool {
+			return *osImage.OpenshiftVersion == versionKey
+		})
+	}
+
+	if osImage == nil {
+		// Find latest available patch version by x.y version
+		osImages := funk.Filter(osImages, func(osImage *models.OsImage) bool {
+			imageVersionKey, err := h.getKey(*osImage.OpenshiftVersion)
+			if err != nil {
+				return false
+			}
+			return imageVersionKey == versionKey
+		}).([]*models.OsImage)
+		sort.Slice(osImages, func(i, j int) bool {
+			return *osImages[i].OpenshiftVersion > *osImages[j].OpenshiftVersion
+		})
+		if !funk.IsEmpty(osImages) {
+			osImage = osImages[0]
 		}
 	}
 
-	return nil, errors.Errorf("The requested OS image for version (%s) isn't specified in OS images list", versionKey)
+	if osImage != nil {
+		return osImage.(*models.OsImage), nil
+	}
+
+	return nil, errors.Errorf(
+		"The requested OS image for version (%s) and CPU architecture (%s) isn't specified in OS images list",
+		openshiftVersion, cpuArchitecture)
 }
 
 // Returns the ReleaseImage entity
 func (h *handler) GetReleaseImage(openshiftVersion, cpuArchitecture string) (*models.ReleaseImage, error) {
-	versionKey, err := h.getKey(openshiftVersion)
-	if err != nil {
-		return nil, err
+	if cpuArchitecture == "" {
+		// Empty implies default CPU architecture
+		cpuArchitecture = common.DefaultCPUArchitecture
 	}
-
-	for _, releaseImage := range h.releaseImages {
-		if cpuArchitecture == "" {
-			// Empty implies default CPU architecture
-			cpuArchitecture = common.DefaultCPUArchitecture
+	// Filter Release images by specified CPU architecture
+	releaseImages := funk.Filter(h.releaseImages, func(releaseImage *models.ReleaseImage) bool {
+		if swag.StringValue(releaseImage.CPUArchitecture) == "" {
```

Comment:
Do we really expect that we would configure the service with a release image with no arch? I don't think this was the case before.